### PR TITLE
E2E & Dashboard

### PR DESCRIPTION
**Pull request by guanbinrui**
This PR is the replacement of #887

### CODE REVIEW

File: .github/workflows/E2E.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    pull_request:
+        branches: [master, released]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+    main:
+        runs-on: self-hosted
+
+        container:
+            image: differui/node-13.2.0-browsers:v2
```
Comment (Jack-Works) [inline .github/workflows/E2E.yml:17]: should update to node 14

File: config-overrides/manifest.overrides.js [hunk 2]
```diff
@@ -29,7 +29,30 @@ function WKWebview(manifest) {
...
+}
+/**
+ * @param {typeof base} manifest
+ */
+function E2E(manifest) {
+    // can not capture premission dialog in pptr
+    manifest.permissions.push(
+        ...manifest.optional_permissions,
+        ...[
+            'https://twitter.com/*',
```
Comment (Jack-Works) [inline config-overrides/manifest.overrides.js:48]: replace with `<all_urls>`?

File: src/social-network-provider/facebook.com/ui-provider.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ export const facebookUISelf = defineSocialNetworkUI({
         facebookUISelf.requestPermission().then((granted) => {
             if (granted) {
                 setStorage('facebook.com', { forceDisplayWelcome: true })
-                location.href = 'https://facebook.com/'
+                location.assign('https://www.facebook.com/')
```
Comment (Jack-Works) [inline src/social-network-provider/facebook.com/ui-provider.ts:52]: what's the difference?

Comment (guanbinrui) [reply src/social-network-provider/facebook.com/ui-provider.ts:52]: It's same.

File: e2e/commands/twitter.ts [hunk 4]
```diff
@@ -153,6 +153,18 @@ class Twitter implements SNS {
         await confirmButton.click()
         await page.waitFor('[href="/settings/profile"]') // confirm back to profile page
     }
+
+    async dimissAnyDialog(page: Page, maxRetries = 3) {
+        for await (const i of new Array(maxRetries).fill(0)) {
+            // click any close buttons on the page
+            const closeButton = await page.$('[role="button"][aria-label="Close"]')
```
Comment (Jack-Works) [inline e2e/commands/twitter.ts:160]: interesting, what "bot dialog"?

**Pull request merged**